lib/youtube/channel-mapper.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * YouTube Channel Data Mapping Utilities
+ * 
+ * Provides reusable functions to transform raw YouTube channel info objects
+ * into standardized channel data structures, eliminating code duplication.
+ */
+
+import { 
+  YouTubeChannelData, 
+  RawChannelInfo, 
+  ChannelInfoResult,
+  YouTubeAccessResult,
+  YouTubeChannelInfoResult
+} from "@/types/youtube";
+
+// Union type for thumbnail formats
+type ThumbnailSource = {
+  default?: string | { url?: string | null } | null;
+  medium?: string | { url?: string | null } | null;
+  high?: string | { url?: string | null } | null;
+} | undefined;
+
+// Union type for sources that might have statistics in different formats
+type StatisticsSource = {
+  subscriberCount?: string;
+  videoCount?: string;
+  viewCount?: string;
+  statistics?: {
+    subscriberCount?: string;
+    videoCount?: string;
+    viewCount?: string;
+  };
+};
+
+// Union type for base channel data sources
+type BaseChannelSource = {
+  id?: string;
+  name?: string;
+  title?: string;
+  description?: string;
+  customUrl?: string | null;
+  country?: string | null;
+  publishedAt?: string | null;
+};
+
+/**
+ * Helper function to normalize thumbnails from various formats to standardized format
+ */
+function normalizeThumbnails(thumbnails: ThumbnailSource) {
+  if (!thumbnails) {
+    return {
+      default: { url: null },
+      medium: { url: null },
+      high: { url: null },
+    };
+  }
+
+  // Handle flat thumbnail format (string URLs)
+  if (typeof thumbnails.default === 'string' || thumbnails.default === null) {
+    return {
+      default: { url: thumbnails.default || null },
+      medium: { url: (thumbnails.medium as string) || null },
+      high: { url: (thumbnails.high as string) || null },
+    };
+  }
```

Comment:
each `file` should export exactly one `function`
- actual: 2+ functions exported in this file
- required: new `file` for each standalone `function`